flink-libraries/flink-table/src/test/scala/org/apache/flink/table/utils/UserDefinedTableFunctions.scala

```diff
@@ -232,7 +244,8 @@ class DynamicSchemaWithRexNodes extends TableFunction[Row] {
       BasicTypeInfo.DOUBLE_TYPE_INFO,
       BasicTypeInfo.DOUBLE_TYPE_INFO,
       BasicTypeInfo.DOUBLE_TYPE_INFO,
-      BasicTypeInfo.BOOLEAN_TYPE_INFO
+      BasicTypeInfo.BOOLEAN_TYPE_INFO,
+      SqlTimeTypeInfo.TIMESTAMP
     )
   }
```

Comment:
Can you add another test table function that determines the result type based on the passed type information?